dbux-data/src/DataProviderBase.js

```diff
@@ -450,7 +450,16 @@ export default class DataProviderBase {
     for (const collectionName in collections) {
       const collection = this.collections[collectionName];
       if (collection.deserialize) {
-        collections[collectionName] = collections[collectionName].map(obj => collection.deserialize(obj));
+        collections[collectionName] = await Promise.all(collections[collectionName].map(obj => collection.deserialize(obj)));
```

Comment:
💥Don't create one promise per object because that could absolutely kill performance.